### PR TITLE
Change return type of `WcsGeom.get_coord()` to quantities

### PR DESCRIPTION
**Pull request by adonath**
This PR addresses #1792. It includes the following changes:

- Change `MapAxis.pix_to_coord()` to return a `Quantity`
- Change `WcsGeom.pix_to_coord()` to return `Quantities`
- Allow `lon` and `lat` to be a `Quantity`in `MapCoord.__init__`
- Adapt tests and notebooks

Shall we declare `lon` and `lat` as `Longitude` and `Latitude` objects instead of `Quantity`?

Currently the adaptation of `HpxGeom.coord_to_pix()` is missing, but if I remember correctly there were problems with the adaptation, when I already tried it ~2 years ago, so I'd prefer to do this in a second PR.

### CODE REVIEW

File: tutorials/image_fitting_with_sherpa.ipynb [hunk 1]
```diff
@@ -430,13 +430,21 @@
...
     "\n",
     "table = Table(rows=rows, names=rows[0])\n",
     "for name in table.colnames:\n",
-    "    table[name].format = \".5g\"\n",
-    "print(table)"
+    "    table[name].format = \".5g\""
+   ]
+  },
+  {
+   "cell_type": "code",
```
Comment (cdeil) [inline tutorials/image_fitting_with_sherpa.ipynb]: Try `make clean-nb`?

**Comment by cdeil** [comment]
This is great!

Maybe spend an hour or two trying to find code in Gammapy where MapCoords are used and see if they can be simplified now?

At https://nbviewer.jupyter.org/github/gammapy/gammapy/blob/46cea1765477f22fd827565e17124b685af9256a/tutorials/fermi_lat.ipynb#Counts I see a comment

>       # The coord-based interface doesn't use Quantity,
>        # so we need to pass energy in the same unit as
>        # used for the map axis

Can this be deleted or does it still apply?

Not sure how to find those points in the code, probably grepping for "deg" yields too much to review.

Long-term, I think that we should consider changing to `rad` instead of `deg` for sky coords.
The WCS interface is deg based, so that's why we have deg at the moment, and that's what's efficient there.
But WCS calls are only made in the setup phase, and then in the model and likelihood loop it's mainly `np.sin` and `np.cos` calls, and I think (but didn't benchmark) that those are a bit faster for "rad", because that's what they use internally, i.e. if we use "deg" they convert to "rad" on every call (which is surprisingly cheap, but still, I expect it's an overhead).

**Comment by cdeil** [comment]
You could also try changing coords.lon and coords.lat from "deg" to "rad", and see if tests / notebooks fail. Those would be places to fix up.

**Comment by adonath** [comment]
Thanks @cdeil! I'll merge the PR now (because it's green) and address you comments in a follow-up PR. This way we avoid merge conflicts in `gammapy/image/models/tests`.

**Pull request merged**